### DISCUSSION TITLE
엑세스 토큰 발급 문의

### DISCUSSION THREAD

**Discussion by ikseon1026** [commerce-api-naver/commerce-solutions#38] [묻고 답하기]
안녕하세요.
판매자가 솔루션 사용신청을 이용하는 것을 가정하여, 개발사 자체몰에 테스트를 하고 있습니다.
아래와 같이 엑세스토큰을 발급하면 정상적으로 발급이 이뤄집니다.

**[방법1] client_id={{솔루션 애플리케이션 ID}}&timestamp={{api_timestamp}}&client_secret_sign={{api_signature}}&grant_type=client_credentials&type=SELF**

하지만, 판매자가 솔루션 사용하는 것을 가정하여 아래의 방식으로 토큰 발급 요청을 하면 
**[방법2] c[lient_id={{솔루션 애플리케이션 ID}}&timestamp={{api_timestamp}}&client_secret_sign={{api_signature}}&grant_type=client_credentials&type=SELLER&account_id={{판매자 UID}}**

다음과 같은 응답이 빌생하면서 토킨이 발급되지 않습니다.

{'code': 'InvalidInputData', 'message': '입력한 데이터가 유효하지 않습니다.', 'timestamp': '2024-10-22T05:51:57.869713Z', 'invalidInputs': [{'name': 'type', 'type': 'NotValid', 'message': 'type 항목이 유효하지 않습니다.'}]}

[판매자 UID] 확인은  솔루션 사용 신청시에 전달되는 JWT를 디코딩 하여 accountUid 를 추출하여 인증 토큰 발급을 싱청하였습니다.
한 번 확인 부탁드리도록 하겠습니다.

감사합니다.

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

인증 토큰 발급 과정에서 사용하시는 애플리케이션에 따라 내부적인 데이터 검증과 동작에 차이가 있습니다.
전달주신 내용만으로 보았을 때 인증 토큰 발급을 시도하시면서 사용하신 애플리케이션이 '솔루션 애플리케이션'이 아닐 가능성이 매우 큰 것으로 보입니다.

혹시 대상 호출건에서 애플리케이션의 ID와 Secret을 솔루션 생성 시 제공받은 '솔루션 애플리케이션' 정보로 활용하셨는지 다시 한 번 확인 부탁드리겠습니다.

**Reply by ikseon1026** [reply]
안녕하세요.  부족하고 제한된 정보임에도 빠르고 정확한 답변 감사합니다.
'솔루션 애플리케이션' 정보 토큰 발급에 성공하였습니다. 고맙습니다.
